opensrp-reveal/src/main/java/org/smartregister/reveal/interactor/ListTaskInteractor.java

```diff
@@ -366,4 +370,14 @@ public void run() {
 
         appExecutors.diskIO().execute(runnable);
     }
+
+    private void saveMosquitoCollectionForm(JSONObject jsonForm) {
+        try {
+            org.smartregister.domain.db.Event event = saveEvent(jsonForm, MOSQUITO_COLLECTION_EVENT, STRUCTURE);
```

Comment:
Process this on background thread and then invoke callback on the UI thread

Reply:
Done.